### DISCUSSION TITLE
Visual toolbar setup and GUI menu editing

### DISCUSSION THREAD

**Discussion by TinchoteSK** [WinEdt-Team/WinEdt-Support#18] [1.1 General Help and Discussion]
After two decades of using WinEdt 5.3, I'm considering updating to version 11. I have downloaded and installed the trial version.  

While checking how to implement all the customizations I have in my old version, it seems as in WinEdt 11, there is no more Visual Toolbar Setup, nor the GUI interface for menu editing. All my attempts took me to the Options Interface, where the only possible way seems to manually edit text configuration files.

Is that the case? Does all WinEdt 11 customization need to be done by editing text configuration files?

**Comment by Alex-Simonic** [comment]
This is correct. If you want to customize the menu, toolbar, etc..., you have to modify and load the configuration script using the Options Interface. There are several advantages to this method of modifying the settings as compared to the old WinEdt 5 quasi-visual interfaces scattered in too many dialogs. Among other things, you can add comments to keep track of your changes for future reference. At first, it may look complicated but once you get a grip on it it should be easy to make any modifications that you have used in older versions (and more).

In the Options menu, you will find a few predefined choices for the toolbar. Pick the one that is the closest to your idea of what the toolbar should look like and then proceed with customizations. If you need guidance with a particular task you can explain and ask for further instructions.

**Reply by TinchoteSK** [reply]
Thanks for the quick reply, Alex. In my experience, I will not edit a long text file without introducing errors. Is there any kind of debugging solution included in this? What happens when one corrupts a configuration script?

**Reply by Alex-Simonic** [reply]
In the Options Interface, you are always modifying a local instance of the configuration file that is automatically copied into your local folder. In most cases, the Load Script command will report and highlight the error and you can use Undo. Right-clicking on the configuration script in the Options Interface tree allows you to Load or Open the default (read-only) instance of that script. In the Options Maintenance menu, you can rebuild default or local settings. Load frequently to catch errors before they get out of hand:-)

While editing configurations scripts you can double-click on many keywords (like SHORTUTS="", IMAGE="", etc) and this will activate a GUI where you can enter the parameter. WinEdt's Help explains the meaning of all keywords and other things you should know before embarking on extensive modifications. I understand that the process may look overwhelmingly complicated and it may not be for everyone. Using common sense and doing some homework by consulting the documentation will help you started.
